src/app/forms/sensor/sensor.component.ts

```diff
@@ -0,0 +1,310 @@
+  }
+
+  public goToStep(step: number): void {
+    if (Object.values(this.formControlSteps)[this.activeStepIndex].some((f) => f.invalid)) {
+      if (step <= this.activeStepIndex) {
+        this.activeStepIndex = step;
+      } else {
+        this.submitted = true;
+      }
+    } else {
+      this.submitted = false;
+      this.activeStepIndex = step;
+    }
+  }
+
+  public getStepCount(): number {
+    return Object.keys(this.formControlSteps).length;
+  }
+
+  public getStepLabel(): string {
+    return Object.keys(this.formControlSteps)[this.activeStepIndex];
+  }
+
+  public getStepClasses(pageIndex: number) {
+    return { active: this.activeStepIndex === pageIndex, finished: this.activeStepIndex > pageIndex };
+  }
+
+  public setSensor(sensor: ISensor): void {
+    this.sensor = sensor;
+
+    this.form.patchValue({
+      aim: this.sensor.aim || '',
+      description: this.sensor.description || '',
+      manufacturer: this.sensor.manufacturer || '',
+      active: { value: this.sensor.active.toString() || false },
+      documentationUrl: this.sensor.documentationUrl || '',
+      location: {
+        longitude: this.sensor.location ? this.sensor.location.coordinates[0] : null,
+        latitude: this.sensor.location ? this.sensor.location.coordinates[1] : null,
+        height: this.sensor.location ? this.sensor.location.coordinates[2] : null,
+        baseObjectId: this.sensor.baseObjectId || 'non-empty',
+      },
+      type: {
+        typeName: this.sensor.typeName ? this.sensor.typeName[0] : '',
+        typeDetails: this.sensor.typeDetails ? this.sensor.typeDetails.subType : '',
+      },
+      theme: { value: this.sensor.theme || [] },
+    });
+
+    // Overwrite the name generated by type.
+    this.form.patchValue({
+      name: this.sensor.name || '',
+    });
+
+    const dataStreams = this.form.get('dataStreams') as FormArray;
+    dataStreams.clear();
+    for (const dataStream of this.sensor.dataStreams) {
+      dataStreams.push(this.formBuilder.group({
+        dataStreamId: dataStream.dataStreamId,
+        name: [dataStream.name, Validators.required],
+        reason: dataStream.reason || '',
+        description: dataStream.description || '',
+        observedProperty: dataStream.observedProperty || '',
+        isPublic: !!dataStream.isPublic,
+        isOpenData: !!dataStream.isOpenData,
+        isReusable: !!dataStream.isReusable,
+        documentationUrl: [dataStream.documentationUrl || '', [Validators.pattern(this.urlRegex)]],
+        dataLink: [dataStream.dataLink || '', [Validators.pattern(this.urlRegex)]],
+        unitOfMeasurement: dataStream.unitOfMeasurement || '',
+        dataFrequency: dataStream.dataFrequency || 0,
+        dataQuality: dataStream.dataQuality || 0,
+      }));
+    }
+  }
+
+  public async submit() {
+    this.submitted = true;
+
+    // Stop if the form is invalid.
+    if (this.form.valid) {
+      if (this.sensor) {  // Update the sensor.
```

Comment:
De submit functie is best omvangrijk geworden. Wil je dit refactoren en aparte functies voor het aanmaken en updaten van een sensor?